ts/store/actions/messages.ts

```diff
@@ -44,9 +44,16 @@ export const setMessageReadState = createAction(
   resolve => (id: string, read: boolean) => resolve({ id, read }, { id, read })
 );
 
+export const setMessagesArchivedState = createAction(
+  "MESSAGES_SET_ARCHIVED",
+  resolve => (ids: ReadonlyArray<string>, archived: boolean) =>
```

Comment:
instead of a `boolean`, I would encode the actual states with and ADT, e.g. : 

```suggestion
  resolve => (ids: ReadonlyArray<string>, state: "VISIBLE" | "ARCHIVED") =>
```

much clearer and less error prone than a boolean

Reply:
I have a diferrent opinion about this for a couple of reason:

1) "visibile" is not an alternative of "archived". i mean what is visible in a screen can be not visibile in onother. The MessageSearchResult component (coming) display both "visible" and "archived" messages (the same is true for the Deadlines)
2) a boolean as representation of the archived state is more like the one that in the future will be implemented server side (i mean the message object returned by the api will have a archived boolean property)
3) if tomorrow we want to add a new state like "starred" seem unreal to have "visible" | "archived" | "starred"

What you think?